dotnet/samples/KernelSyntaxExamples/Example07_BingAndGoogleSkills.cs

```diff
@@ -139,26 +142,32 @@ private static async Task Example2Async(IKernel kernel)
 
         var oracle = kernel.CreateSemanticFunction(SemanticFunction, maxTokens: 200, temperature: 0, topP: 1);
 
-        var context = kernel.CreateNewContext();
-        context.Variables["externalInformation"] = "";
-        var answer = await oracle.InvokeAsync(questions, context);
+        var answer = await kernel.RunAsync(oracle,
+            input: questions,
```

Comment:
I think separate optional `input` parameter is not a problem, it will simplify the usage in case there is only one input parameter. We can support both scenarios at the same time if needed.

Just as idea: we can also provide a signature where args are passed as `params`, in similar way how `string.Format` works. In this case the usage will be easier comparing to `Dictionary` initialization syntax. In case of `params` it won't be possible to map parameter key to parameter value, but values can be applied to keys by following the same order how they are defined in prompt and how they are passed to the method (like in `logger.LogInformation(str, params)`).

Reply:
I agree, and would really rather not initialize the new Dictionary.

params is a neat trick, but unfortunately gets in the way of passing the CancellationToken as a final arg (see IKernel.RunAsync overloads).
It looks really weird when you use params, and have CancellationToken in the middle of the method signature.

I have an idea to try out here... stay tuned.

Reply:
Cleaned up.